src/xray/jest-xray-environment.ts

```diff
@@ -0,0 +1,37 @@
+import NodeEnvironment from "jest-environment-node";
+
+class JestXRayEnvironment extends NodeEnvironment {
+
+  override async setup() {
+    await super.setup();
+  }
+
+  async handleTestEvent(event: any) {
+    if (event.name === "test_start") {
+      let testNames = [];
+      let currentTest = event.test;
+
+      while (currentTest) {
+        testNames.push(currentTest.name);
+        currentTest = currentTest.parent;
+      }
+
+      if (typeof testNames[0] === 'string') {
+        testNames[0] = testNames[0].trim();
+      }
+      if (typeof testNames[1] === 'string') {
+        testNames[1] = testNames[1].trim();
+      }
+      this.global["testName"] = testNames[0]
+      this.global["describeName"] = testNames[1]
```

Comment:
Don't understand why we need the trimming but wouldn't it be easier to have?
```suggestion
      this.global["testName"] = testNames.lenght > 0 ? testNames[0].trim() : undefined;
      this.global["describeName"] =  testNames.lenght > 1 ? testNames[1].trim() : undefined;
```